### PR TITLE
Issue #1149 Checks for memory size

### PR DESCRIPTION
**Pull request by gbraad**
Addresses issue #1149

Just for a quick review...

Some things of note:
  * When defining memory as 4G -> 4 GB is displayed
  * When 4000M is used -> 3 GB is displayed
  * Texts have not been altered yet

### CODE REVIEW

**Comment by gbraad** [comment]
```
func calculateMemorySize(memorySize string) int {
...
    // RAMInBytes("5G") => 5368709120 / units.MiB = 5120
    size, err := units.RAMInBytes(memorySize)
    return int(size / units.MiB)
```

```
func calculateMemorySize(memorySize string) int {
...
    // FromHumanSize("5G") => (5000000000 / units.GB) * units.KiB = 5120
    size, err := units.FromHumanSize(memorySize)
    return int((size / units.GB) * units.KiB)
```

File: cmd/minishift/cmd/start.go [hunk 1]
```diff
@@ -411,18 +411,34 @@ func exportContainerImages(hostVm *host.Host, version string) {
...
 
-// calculateDiskSizeInMB converts a human specified disk size like "1000MB" or "1GB" and converts it into Megabits
-func calculateDiskSizeInMB(humanReadableSize string) int {
+func calculateMemorySize(memorySize string) int {
+	if stringUtils.HasOnlyNumbers(memorySize) {
+		memorySize += "MB"
+	}
+
+	// err := minishiftConfig.IsValidMemorySize(configCmd.Memory.Name, humanReadableSize)
+	size, err := units.RAMInBytes(memorySize)
```
Comment (gbraad) [inline cmd/minishift/cmd/start.go]:
```go
func calculateMemorySize(memorySize string) int {
...
    // RAMInBytes("5G") => 5368709120 / units.MiB = 5120
    size, err := units.RAMInBytes(memorySize)
    return int(size / units.MiB)
```

Comment (gbraad) [inline cmd/minishift/cmd/start.go]:
```go
func calculateMemorySize(memorySize string) int {
...
    // FromHumanSize("5G") => (5000000000 / units.GB) * units.KiB = 5120
    size, err := units.FromHumanSize(memorySize)
    return int((size / units.GB) * units.KiB)
```

File: cmd/minishift/cmd/start.go [hunk 2]
```diff
@@ -411,18 +411,34 @@ func exportContainerImages(hostVm *host.Host, version string) {
 	fmt.Println(fmt.Sprintf("-- Exporting of OpenShift images is occuring in background process with pid %d.", exportCmd.Process.Pid))
 }
 
-// calculateDiskSizeInMB converts a human specified disk size like "1000MB" or "1GB" and converts it into Megabits
-func calculateDiskSizeInMB(humanReadableSize string) int {
+func calculateMemorySize(memorySize string) int {
+	if stringUtils.HasOnlyNumbers(memorySize) {
+		memorySize += "MB"
```
Comment (gbraad) [inline cmd/minishift/cmd/start.go]: This is for backwards compatibility with the currently used values.

**Comment by hferentschik** [comment]
@gbraad is this still WIP?

**Comment by gbraad** [comment]
Can be reviewed nonetheless, as the main functionality has been implemented.

**Comment by hferentschik** [comment]
Ok

**Comment by gbraad** [comment]
`HumanSizeWithPrecision` Is not in any tagged version :-(
https://github.com/docker/go-units/blob/0dadbb0345b35ec7ef35e228dabb8de89a65bf52/size.go#L57-L62

File: cmd/minishift/cmd/config/config.go [hunk 3]
```diff
@@ -50,7 +50,7 @@ var (
 	// minishift
 	ISOUrl           = createConfigSetting("iso-url", SetString, []setFn{validations.IsValidUrl}, []setFn{RequiresRestartMsg}, true)
 	CPUs             = createConfigSetting("cpus", SetInt, []setFn{validations.IsPositive}, []setFn{RequiresRestartMsg}, true)
-	Memory           = createConfigSetting("memory", SetInt, []setFn{validations.IsPositive}, []setFn{RequiresRestartMsg}, true)
+	Memory           = createConfigSetting("memory", SetString, []setFn{validations.IsValidMemorySize}, []setFn{RequiresRestartMsg}, true)
```
Comment (hferentschik) [inline cmd/minishift/cmd/config/config.go]: :+1

File: cmd/minishift/cmd/start.go (hunk 2 above)
Comment (hferentschik) [inline cmd/minishift/cmd/start.go]: Got you.

File: cmd/minishift/cmd/start.go [hunk 4]
```diff
@@ -449,9 +466,9 @@ func initStartFlags() *flag.FlagSet {
 
 	startFlagSet.String(configCmd.ISOUrl.Name, minishiftConstants.B2dIsoAlias, "Location of the minishift ISO. Can be an URL, file URI or one of the following short names: [b2d centos].")
 	startFlagSet.String(configCmd.VmDriver.Name, constants.DefaultVMDriver, fmt.Sprintf("The driver to use for the Minishift VM. Possible values: %v", constants.SupportedVMDrivers))
-	startFlagSet.Int(configCmd.Memory.Name, constants.DefaultMemory, "Amount of RAM to allocate to the Minishift VM.")
 	startFlagSet.Int(configCmd.CPUs.Name, constants.DefaultCPUS, "Number of CPU cores to allocate to the Minishift VM.")
-	startFlagSet.String(configCmd.DiskSize.Name, constants.DefaultDiskSize, "Disk size to allocate to the Minishift VM. Use the format <size><unit>, where unit = b, k, m or g.")
+	startFlagSet.String(configCmd.Memory.Name, constants.DefaultMemory, "Amount of RAM to allocate to the Minishift VM. Use the format <size><unit>, where unit = MB or GB.")
```
Comment (hferentschik) [inline cmd/minishift/cmd/start.go]: +1 ahh, MB and GB. Very nice.

**Comment by hferentschik** [comment]
Rebased and merged, thanks.

**Comment by jorgemoralespou** [comment]
How is this feature used? What vaues and what is the valid formats? I don't see any documentation update.

**Comment by gbraad** [comment]
documentation is included in the generated flag options:

```
"Amount of RAM to allocate to the Minishift VM. Use the format <size><unit>, where unit = MB or GB."
```

**Pull request closed without merging**